diff --git a/index.js b/index.js
index 23384ea16bb848616886f356d8d69f94325535bf..66c157aeb94d23a787accc87e1f0f94328cef81a 100644
--- a/index.js
+++ b/index.js
@@ -126,6 +126,16 @@ class Extractor {
 
     if (symlink) {
       const link = await getStream(readStream)
+      const target = path.resolve(path.dirname(dest), link)
+      const relativeTarget = path.relative(this.opts.dir, target)
+      if (
+        path.posix.isAbsolute(link) ||
+        path.win32.isAbsolute(link) ||
+        path.isAbsolute(relativeTarget) ||
+        relativeTarget.split(path.sep).includes('..')
+      ) {
+        throw new Error(`Out of bound symlink target "${link}" found while processing file ${entry.fileName}`)
+      }
       debug('creating symlink', link, dest)
       await fs.symlink(link, dest)
     } else {
